### PR TITLE
Sudoku base classes

### LINKED ISSUE #1: implement Sudoku Classes and Access Interfaces
- UML (?)

### PR DESCRIPTION
**Pull request by LucRome**
- Add base Classes for the Sudoku
- including tests
    - might be improvable
    -

### CODE REVIEW

File: src/backend/sudoku/base.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    def __init__(self, value: Optional[int] = None):
+        if value is not None:
+            self.set_value(value)
+        else:
+            self.remove_value()
+
+    def set_value(self, val: int) -> None:
+        if val < FIELD_VALUE_MIN or val > FIELD_VALUE_MAX:
+            raise WrongFieldValueExcetion(val)
```
Comment (noahhauke22) [inline src/backend/sudoku/base.py]: WrongFieldValueExcetion -> WrongFieldValueException

File: src/backend/sudoku/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"""
+Contains exceptions
+"""
+
+class WrongFieldValueExcetion(Exception):
```
Comment (noahhauke22) [inline src/backend/sudoku/exceptions.py]: WrongFieldValueExcetion -> WrongFieldValueException

File: src/backend/sudoku/tests.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        val = 3
+        field.set_value(val=val)
+        self.assertEqual(val, field.get_value())
+        field.remove_value()
+        self.assertIsNone(field.get_value())
+        field = Field()
+        self.assertIsNone(field.get_value())
+
+        # throw error
+        with self.assertRaises(WrongFieldValueExcetion):
```
Comment (noahhauke22) [inline src/backend/sudoku/tests.py]: WrongFieldValueExcetion -> WrongFieldValueException

File: src/backend/sudoku/tests.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            field.set_value(val=(FIELD_VALUE_MAX + 1))
+        with self.assertRaises(WrongFieldValueExcetion):
```
Comment (noahhauke22) [inline src/backend/sudoku/tests.py]: WrongFieldValueExcetion -> WrongFieldValueException

**Review by noahhauke22 — APPROVED** [review]
Just a spelling Error at WrongFieldValueExcetion -> WrongFieldValueException

File: src/backend/sudoku/base.py (hunk 1 above)
Comment (LucRome) [reply src/backend/sudoku/base.py]: Fixed

File: src/backend/sudoku/tests.py (hunk 4 above)
Comment (LucRome) [reply src/backend/sudoku/tests.py]: Fixed

**Pull request merged**